docs/content/drafts/PageLayout.mdx

```diff
@@ -0,0 +1,182 @@
+---
+title: PageLayout
+status: Draft
+description: TODO
+---
+
+<Note variant="warning">Not implemented yet</Note>
+
+## Examples
+
+### Default
+
+```jsx
+<PageLayout>
+  <PageLayout.Header>Header</PageLayout.Header>
+  <PageLayout.Content>Content</PageLayout.Content>
+  <PageLayout.Pane>Pane</PageLayout.Pane>
+  <PageLayout.Footer>Footer</PageLayout.Footer>
+</PageLayout>
+```
+
+### With dividers
+
+```jsx
+<PageLayout>
+  <PageLayout.Header divider="line">Header</PageLayout.Header>
+  <PageLayout.Content>Content</PageLayout.Content>
+  <PageLayout.Pane divider="line">Pane</PageLayout.Pane>
+  <PageLayout.Footer divider="line">Footer</PageLayout.Footer>
+</PageLayout>
+```
+
+### Condensed
+
+```jsx
+<PageLayout columnGap="condensed" rowGap="condensed">
+  <PageLayout.Header>Header</PageLayout.Header>
+  <PageLayout.Content>Content</PageLayout.Content>
+  <PageLayout.Pane>Pane</PageLayout.Pane>
+  <PageLayout.Footer>Footer</PageLayout.Footer>
+</PageLayout>
+```
+
+## Props
+
+### PageLayout
+
+<!-- TODO: Resoponsive variants -->
+
+<PropsTable>
+  <PropsTableRow
+    name="containerWidth"
+    type={`| 'full'
+| 'medium'
+| 'large'
+| 'xlarge'`}
+    defaultValue="'full'"
+    description="The maximum width of the page container."
+  />
+  <PropsTableRow
+    name="outerSpacing"
+    type={`| 'normal'
+| 'condensed'`}
+    defaultValue="'normal'"
+    description="The spacing between the outer edges of the page container and the viewport"
+  />
+  <PropsTableRow
+    name="columnGap"
+    type={`| 'normal'
+| 'condensed'`}
+    defaultValue="'normal'"
+  />
+  <PropsTableRow
+    name="rowGap"
+    type={`| 'normal'
+| 'condensed'`}
+    defaultValue="'normal'"
+  />
+  <PropsTableSxRow />
+</PropsTable>
+
+### PageLayout.Header
+
+<PropsTable>
+  <PropsTableRow name="divider" type="'none' | 'line'" defaultValue="'none'" />
+  <PropsTableRow
+    name="dividerWhenNarrow"
+    type={`| 'inherit'
+| 'none'
+| 'line'
+| 'filled'`}
+    defaultValue="'inherit'"
+  />
+  <PropsTableSxRow />
+</PropsTable>
+
+### PageLayout.Content
+
+<PropsTable>
+  <PropsTableRow
+    name="width"
+    type={`| 'full'
+| 'medium'
+| 'large'
+| 'xlarge'`}
+    defaultValue="'full'"
+    description="Define the maximum width of the content region."
+  />
+  <PropsTableSxRow />
+</PropsTable>
+
+### PageLayout.Pane
+
+<PropsTable>
+  <PropsTableRow
+    name="position"
+    type={`| 'start'
+| 'end'`}
+    defaultValue="'start'"
```

Comment:
Hi! Because we're in composition land, how do you feel about not having a position prop at all and rendering them in the order they are written? 😳

```jsx
// right sidebar = content first, pane after
<PageLayout>
  <PageLayout.Header>Header</PageLayout.Header>
  <PageLayout.Content>Content</PageLayout.Content>
  <PageLayout.Pane>Pane</PageLayout.Pane>
</PageLayout>

// and left sidebar = pane first, content after
<PageLayout>
  <PageLayout.Header>Header</PageLayout.Header>
  <PageLayout.Pane>Pane</PageLayout.Pane>
  <PageLayout.Content>Content</PageLayout.Content>
</PageLayout>

// sidebar right on wide, top on narrow?
<PageLayout>
  <PageLayout.Header>Header</PageLayout.Header>
  <PageLayout.Content>Content</PageLayout.Content>
  <PageLayout.Pane positionWhenNarrow="start">Pane</PageLayout.Pane>
</PageLayout>

// ⚠️ entering weird:
// using positionWhenNarrow to reverse the order feels like `flex-direction: row-reverse`
// should we give into that and put it on the parent? 🤔 
// implicit assumption that it wouldn't flip the order of header and footer?
<PageLayout narrowVariant="reverse">
  <PageLayout.Header>Header</PageLayout.Header>
  <PageLayout.Content>Content</PageLayout.Content>
  <PageLayout.Pane>Pane</PageLayout.Pane>
</PageLayout>
```

Reply:
@siddharthkp since the contents of `<PageLayout.Pane>` and `<PageLayout.Content>` can grow a lot (they host the entire page structure after all), I'd say it's more readable to see the pane position specified as a prop directly.

Also, header and footer should always appear at the top and at the bottom, respectively, regardless of the order they're written in 🤔 .

Reply:
I can see that happening 🤔 

they might even end up in different files, sold!

Reply:
Agreed. Let's move forward with the `position` prop for now 👍